test/e2e/kappcontroller/app_secret_configmap_reconcile_test.go

```diff
@@ -209,3 +209,18 @@ data:
 		})
 	})
 }
+
+func retry(t *testing.T, timeout time.Duration, f func() error) {
+	var err error
+	stopTime := time.Now().Add(timeout)
```

Comment:
fwiw i'd vote for counting retry iterations over measuring time -  there's several cases where time can move around on you,  for instance if we're running on a virtual host which suspends and resumes,  or if the host clock has skewed and during our test run it resyncs with NTP... ( see also,  https://gist.github.com/timvisee/fcda9bbdff88d45cc9061606b4b923ca )

Reply:
cool: https://github.com/vmware-tanzu/carvel-kapp-controller/issues/351